src/os/pl-stream.c

```diff
@@ -2101,9 +2113,15 @@ next_chr(const char **s, IOENC enc)
   }
 }
 
+#ifdef O_DEBUG
+#define OUTCHR(s, c)	do { printed++; last_char = (c); \
```

Comment:
I think you missed that streams already have an `s->lastc` field.   That should simplify things a little, no?

Reply:
Aha, I absolutely did miss that! Fixing, please stand by...